ui/tests/e2e/extensions.spec.js

```diff
@@ -0,0 +1,133 @@
+    // Verify Kanvas heading using data-testid
+    await expect(page.getByTestId('kanvas-signup-heading')).toBeVisible();
+
+    const [newPage] = await Promise.all([
+      context.waitForEvent('page'),
+      page.getByTestId('kanvas-signup-btn').click(),
+    ]);
+
+    await expect(newPage).toHaveURL('https://docs.layer5.io/kanvas/');
+    await newPage.close();
+  });
+
+  test('Verify Meshery Docker Extension Details', async ({ page, context }) => {
+    // Verify Docker Extension heading using data-testid
+    await expect(page.getByTestId('docker-extension-heading')).toBeVisible();
+
+    const [newPage] = await Promise.all([
+      context.waitForEvent('page'),
+      page.getByTestId('docker-extension-download-btn').click(),
+    ]);
+
+    await expect(newPage).toHaveURL(
+      'https://hub.docker.com/extensions/meshery/docker-extension-meshery',
+    );
+    await newPage.close();
+  });
+
+  test('Verify Meshery Design Embed Details', async ({ page, context }) => {
+    await expect(page.getByTestId('design-embed-learn-more-btn')).toBeVisible();
+
+    const [newPage] = await Promise.all([
+      context.waitForEvent('page'),
+      page.getByTestId('design-embed-learn-more-btn').click(),
+    ]);
+
+    await expect(newPage).toHaveURL('https://docs.layer5.io/kanvas/designer/embedding-designs/');
+    await newPage.close();
+  });
+
+  test('Verify Meshery Catalog Section Details', async ({ page, context }) => {
+    // Verify Catalog section heading using data-testid
+    await expect(page.getByTestId('catalog-section-heading')).toBeVisible();
+
+    // Verify catalog toggle switch using data-testid
+    await expect(page.getByTestId('catalog-toggle-switch')).toBeVisible();
+
+    // First verify the toggle button functionality
+    const toggleButton = page
+      .locator(
+        'div:has-text("Explore the Meshery Catalog") >> .MuiSwitch-root input[type="checkbox"]',
+      )
+      .nth(0);
+    await expect(toggleButton).toBeVisible();
+
+    // Click toggle Button and Verify the State changes
+    const initialState = await toggleButton.isChecked();
+    await toggleButton.click();
+    await expect(toggleButton).toBeChecked(!initialState);
+
+    // Verify the Meshery Catalog Link
+    const catalogLink = page.locator('a[href="https://meshery.io/catalog"]');
+    const [newPage] = await Promise.all([context.waitForEvent('page'), catalogLink.click()]);
+
+    // Verify the URL
+    await expect(newPage).toHaveURL('https://meshery.io/catalog');
+    await newPage.close();
+  });
+
+  test('Verify Meshery Adapter for Istio Section', async ({ page, context }) => {
+    // Find the Istio section container to scope our selectors
+    const istioSection = page.locator('div', {
+      has: page.locator('text=Meshery Adapter for Istio'),
+    });
+
+    // Test the "Open Adapter docs" link
+    const adapterDocsLink = istioSection.locator('a:has-text("Open Adapter docs")').first();
+    await expect(adapterDocsLink).toBeVisible();
```

Comment:
Curious if theres a better locator for this w/ the ids we put it? If not maybe something like this would be cleaner - 
 ```
const adapterDocsLink = page.getByRole('link', { 
    name: 'Open Adapter docs'
  });
```


maybe we even inline this vs a separate variable as the variable name is now redudant w/ a cleaner locator?